crates/interpreter/src/interpreter.rs

```diff
@@ -93,26 +93,45 @@ impl Interpreter {
         }
     }
 
-    /// When sub create call returns we can insert output of that call into this interpreter.
-    pub fn insert_create_output(&mut self, result: InterpreterResult, address: Option<Address>) {
-        self.return_data_buffer = match result.result {
-            // Save data to return data buffer if the create reverted
-            return_revert!() => result.output,
-            // Otherwise clear it
-            _ => Bytes::new(),
-        };
+    /// Inserts the output of a `create` call into the interpreter.
+    ///
+    /// This function is used after a `create` call has been executed. It processes the outcome
+    /// of that call and updates the state of the interpreter accordingly.
+    ///
+    /// # Arguments
+    ///
+    /// * `create_outcome` - A `CreateOutcome` struct containing the results of the `create` call.
+    ///
+    /// # Behavior
+    ///
+    /// The function updates the `return_data_buffer` with the data from `create_outcome`.
+    /// Depending on the `next_action` indicated by `create_outcome`, it performs one of the following:
+    ///
+    /// - `NextAction::Continue`: Pushes the address from `create_outcome` to the stack, updates gas costs, and records any gas refunds.
+    /// - `NextAction::Revert`: Pushes `U256::ZERO` to the stack and updates gas costs.
+    /// - `NextAction::ExternalErr`: Sets the `instruction_result` to `InstructionResult::FatalExternalError`.
+    /// - Other actions: Pushes `U256::ZERO` to the stack.
+    ///
+    /// # Side Effects
+    ///
+    /// - Updates `return_data_buffer` with the data from `create_outcome`.
+    /// - Modifies the stack by pushing values depending on the `next_action`.
+    /// - Updates gas costs and records refunds in the interpreter's `gas` field.
+    /// - May alter `instruction_result` in case of external errors.
+    pub fn insert_create_outcome(&mut self, create_outcome: CreateOutcome) {
+        self.return_data_buffer = create_outcome.data_buffer();
 
-        match result.result {
-            return_ok!() => {
-                push_b256!(self, address.unwrap_or_default().into_word());
-                self.gas.erase_cost(result.gas.remaining());
-                self.gas.record_refund(result.gas.refunded());
+        match create_outcome.next_action() {
```

Comment:
Would leave this unchanged, it looks better, but macros are used so we don't have performance degradation. Would definitely consider this later but not as part of this PR.